### ISSUE TITLE
Fix FP S2189 (`no-infinite-loop `): consider unmodified conditions that are compared

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#2744]
```javascript
let value = /*... */ ;
while (value < foo()) {
  /* ... */
}
```

Reported in https://community.sonarsource.com/t/false-positive-for-indefinite-loop-with-multiple-conditions-javascript-s2189/44970/3

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-128
